### PR TITLE
Prototype/tip endpoint adjustments

### PR DESCRIPTION
**Pull request by thepiwo**

### CODE REVIEW

File: src/components/tipRecords/AuthorAndDate.vue [hunk 1]
```diff
@@ -67,11 +70,6 @@ export default {
         this.hover = hover;
       }, 500),
     },
-    ...mapState({
-      name({ chainNames }) {
```
Comment (davidyuk) [inline src/components/tipRecords/AuthorAndDate.vue:71]: so, `chainNames` doesn't contain all needed names anymore?

File: src/components/tipRecords/TipPreview.vue [hunk 2]
```diff
@@ -183,16 +183,17 @@ export default {
...
+      return this.tip?.LinkPreview?.title || '';
     },
     tipPreviewImage() {
-      return this.isPreviewToBeVisualized && this.tip.preview.image !== null ? Backend.getTipPreviewUrl(this.tip.preview.image) : '';
+      return this.isPreviewToBeVisualized && this.tip.LinkPreview.image !== null ? Backend.getTipPreviewUrl(this.tip.LinkPreview.image) : '';
     },
     isPreviewToBeVisualized() {
-      return this.tip.preview && (this.tip.preview.description || this.tip.preview.title);
+      return this.tip.LinkPreview
+        && (this.tip.LinkPreview.description || this.tip.LinkPreview.title);
```
Comment (davidyuk) [inline src/components/tipRecords/TipPreview.vue:196]:
```suggestion
      return this.tip.LinkPreview?.description || this.tip.LinkPreview?.title;
```

File: src/components/tipRecords/TipPreview.vue [hunk 3]
```diff
@@ -112,7 +112,7 @@
             <Avatar :address="tip.receiver" />
             <div class="tip-author_name">
               <span class="chain-name">
-                {{ tip.chainName }}
+                {{ tip.ChainName.name }}
```
Comment (davidyuk) [inline src/components/tipRecords/TipPreview.vue:115]: question to the backend: usually types starts with a capital letter 🤷‍♀️

Comment (thepiwo) [reply src/components/tipRecords/TipPreview.vue:115]: The problem is with the new aggregation and sequelize we are using, for join of different models the model is upper case in the result, for some materialized view results its all lowercase, so excuse that confusion

File: src/components/tipRecords/AuthorAndDate.vue (hunk 1 above)
Comment (thepiwo) [reply src/components/tipRecords/AuthorAndDate.vue:71]: guess so

File: src/components/UserInfo.vue [hunk 4]
```diff
@@ -92,7 +92,7 @@
             :href="explorerTransactionsUrl"
           >
             <div class="chain">
-              {{ userChainName ? userChainName : $t('FellowSuperhero') }}
+              {{ userChainName && userChainName.length ? userChainName[0] : $t('FellowSuperhero') }}
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/components/UserInfo.vue:95]: can't seem to use that syntax in the html part

File: src/components/Avatar.vue [hunk 5]
```diff
@@ -20,23 +19,23 @@ jdenticon.config = IDENTICON_CONFIG;
...
-      const name = chainNames[this.address];
-      return name
-        ? new Avatars(sprites, AVATAR_CONFIG).create(name)
+  computed: {
+    profileIdenticonUrl() {
+      return this.chainName
+        ? new Avatars(sprites, AVATAR_CONFIG).create(this.chainName)
         : `data:image/svg+xml;base64,${btoa(jdenticon.toSvg(this.address, 32))}`;
     },
     profileImageUrl({ address, profile }) {
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/components/Avatar.vue:34]: thanks, still wip :)

[the remark this answers is not in the document]
Comment (thepiwo) [reply src/components/Avatar.vue:34]: actually it does not, its a prop passed

File: src/components/tipRecords/AuthorAndDate.vue (hunk 1 above)
Comment (thepiwo) [reply src/components/tipRecords/AuthorAndDate.vue:71]: they are now all included in the respective backend response, before it was duplicate from state and backend as well

File: src/components/UserInfo.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (davidyuk) [reply src/components/UserInfo.vue:95]: we have to update to vue@3 to solve this 😄 https://github.com/vuejs/vue/issues/11088

File: src/components/Avatar.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (davidyuk) [reply src/components/Avatar.vue:34]: `address` and `profile` arguments are fields of state? As I understand it was provided by `mapState`. Without `mapState` the [first argument](https://vuejs.org/v2/api/#computed) of a computed variable is the current component, but I don't see `profile` variable defined there, also the comparison `address === this.address` doesn't make sense then.

File: src/views/Landing.vue [hunk 6]
```diff
@@ -131,29 +131,29 @@
           >
             <VueAos animation-class="animate__animated animate__delay-1s animate__fadeInUp">
               <div class="sh-col-3 sm-text-box">
-                <h4>{{ stats.total_tips_length }}</h4>
+                <h4>{{ stats.totaltipslength }}</h4>
```
Comment (davidyuk) [inline src/views/Landing.vue:134]: naming gets more unusual 🙃

Comment (thepiwo) [reply src/views/Landing.vue:134]: Yeah... Not what I wish, but the return value from materialized views are forced lowercase

Comment (thepiwo) [reply src/views/Landing.vue:134]: We could map it to different names in backend, we should probably discuss that with the team

Comment (thepiwo) [reply src/views/Landing.vue]: Its fixed now :)

File: src/components/tipRecords/TipPreview.vue (hunk 3 above)
Comment (thepiwo) [reply src/components/tipRecords/TipPreview.vue]: fixed now :)

File: src/components/tipRecords/AuthorAndDate.vue (hunk 1 above)
Comment (thepiwo) [reply src/components/tipRecords/AuthorAndDate.vue]: it's still needed, should be resolved now

File: src/components/UserInfo.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/components/UserInfo.vue]: I'd be in favour of this :D

**Comment by kenodressel** [comment]
Before merging we have to revert 1c02cb07c7b60fc43e3ebc59955abf4cf69ee04f

**Comment by Liubov-crypto** [comment]
Got these errors in the wallet. They appear immediately after connecting to the testnet:
warning
err 1
eerr2

Retip error: 
err3
retip error

Tip comment displaying: 
tip comment displaying

My profile: 
Edit profile info- pass
Change pictures - pass
Cookie policy - pass
Feed:
Pin/Unpin - pass
Report post - pass
Claim post - pass
Comments - pass
Tip comment - pass
Tip on the feed - pass
Retip content on the feed - pass
Post without tipping - pass
Voting:
Vote - pass
Revoke vote - pass
Search poll - pass
Delegate voting power - pass
Create a poll - pass
Meet - pass
Updates - pass
FAQ - pass

Wordbazaar:
Buy tokens - pass
Claimed back tokens - pass
Sell tokens - was unable to sell tokens, the button is not active
Create tokens - the transaction takes a very long time and was not completed
Initiate voting - was unable to initiate voting, the button is not active

Can these errors be related to the fact that other tokens are disabled in the last wallet (commit 1daf171 / v.0.7.1 ) and only ae tokens are available?

**Review by AtanasKrondev — APPROVED** [review]
The PR passes the tests it is related to.
The documented errors are not caused by or related to the changes in the PR, so for me it's ok to proceed.

**Comment by Liubov-crypto** [comment]
Retested with iframe and extension:
My profile: 
 Edit profile info- pass 
Change profile/background pictures - pass 
 Cookie policy - pass 
 Feed:  
Pin/Unpin - pass 
Report post - pass 
Claim post - pass 
 Comments - pass 
Tip comment - pass 
 Tip on the feed - pass 
 Retip content on the feed - pass 
 Post without tipping - pass 
Search/ using hashtags - pass 
 Voting:
 Vote - pass 
 Revoke vote - pass 
 Search poll - pass 
Delegate voting power - pass 
 Create a poll - pass 
Meet - pass 
 Updates - pass 
 FAQ - pass 
Wordbazaar: 
Buy tokens - pass 
 Sell tokens - pass 
 Create tokens - failed (the transaction takes a very long time and was not completed)
 Initiate voting - pass 
Vote - pass 
Revoke - pass  
Claimed back tokens - pass

Token creation was not completed: 
creating token
cr token

**Comment by thepiwo** [comment]
thanks @Liubov-crypto thats probably due to the existing testnet issues, I'll check it out

**Comment by etharner** [comment]
I've checked testnet wallet behaviour on https://prototype-tip-endpoint-adjustments.aggregator.z52da5wt.xyz/
Retip is working
Tip giving up after 10 blocks (probably works too)
Comments is working

**Comment by Liubov-crypto** [comment]
> thanks @Liubov-crypto thats probably due to the existing testnet issues, I'll check it out

Thank you!

**Pull request merged**